src/modules/party/party.controller.ts

```diff
@@ -64,18 +66,40 @@ export class PartyController {
   })
   async createParty(
     @ValidatedArrayBody({ items: CreatePartyRequestItem })
-    createPartyDto: CreatePartyRequest,
+    createPartyDto: CreatePartyRequestDto,
     @Res({ passthrough: true }) res: Response,
   ): Promise<CreatePartyResponse> {
     const [newParty] = createPartyDto;
-    const partyIdentifierOfMatchingParty = await this.partyService.getPartyIdentifierBySearchText(newParty.alternateIdentifier);
+    const { alternateIdentifier, officerRiskDate: ratedDate } = newParty;
+
+    const partyIdentifierOfMatchingParty = await this.partyService.getPartyIdentifierBySearchText(alternateIdentifier);
 
     if (partyIdentifierOfMatchingParty) {
       res.status(HttpStatus.OK);
       return partyIdentifierOfMatchingParty;
     }
 
-    return this.partyService.createParty(newParty);
+    const { partyIdentifier } = await this.partyService.createParty(newParty);
+
+    const existingExternalRatingsOfParty = await this.partyExternalRatingService.getExternalRatingsForParty(partyIdentifier);
+
+    if (Array.isArray(existingExternalRatingsOfParty) && existingExternalRatingsOfParty.length === 0) {
+      // TODO APIM-336: This is a placeholder function which returns a hard-coded value, and should be replaced when Informatica is ready.
+      const [salesforceParty] = await this.salesforceGetPartyByAlternateIdentifierPlaceholder();
+
+      const accountType = salesforceParty?.account?.type;
+      const { SOVEREIGN, CORPORATE } = ENUMS.ASSIGNED_RATING_CODES;
+      const assignedRatingCode = SOVEREIGN_ACCOUNT_TYPES.includes(accountType) ? SOVEREIGN : CORPORATE;
+
+      await this.partyExternalRatingService.createExternalRatingForParty(partyIdentifier, { assignedRatingCode, ratedDate });
+    }
```

Comment:
I think this is too much logic for the controller and it might be better off in a service

Reply:
I've kept the logic in the controller but added helper private method and refactored for clarity, as discussed